src/cmd/flash.js

```diff
@@ -176,6 +213,20 @@ module.exports = class FlashCommand extends CLICommandBase {
 		return JSON.parse(manifest.toString());
 	}
 
+	async _getFirehoseFileFromZip(zipPath) {
+		const dir = await unzip.Open.file(zipPath);
+		const firehoseFile = dir.files.find(file => file.path.includes('firehose'));
+		if (!firehoseFile) {
+			throw new Error('Unable to find firehose file');
+		}
```

Comment:
The logic to find the firehose file is already present in `_extractFlashFilesFromZip` and is more precise: open the zip, load the manifest JSON, get the field for the firehose loader.

Call `_extractFlashFilesFromZip` and get the loader filename from the first element of the returned `files`. Then load that file from the zip.